tpls/runtime-template.js

```diff
@@ -204,5 +204,22 @@ function applyUpdate(callback, errback) {
   <% } %>
 }
 
+function updateOfflineService() {
+  if (navigator.serviceWorker) {
```

Comment:
There is `hasSW()` function which should be used instead of `navigator.serviceWorker` check.